ui/pages/token-allowance/token-allowance.test.js

```diff
@@ -0,0 +1,334 @@
+            value: {
+              estimateType: 'custom',
+              gas: '0xea60',
+              maxFeePerGas: '0x4a817c800',
+              maxPriorityFeePerGas: '0x4a817c800',
+            },
+          },
+        ],
+      ],
+      userFeeLevel: 'custom',
+      defaultGasEstimates: {
+        estimateType: 'custom',
+        gas: '0xea60',
+        maxFeePerGas: '0x4a817c800',
+        maxPriorityFeePerGas: '0x4a817c800',
+        gasPrice: '0x4a817c800',
+      },
+    },
+  };
+
+  let store;
+  beforeEach(() => {
+    store = configureMockStore()(state);
+  });
+
+  it('should render title "Set a spending cap for your" in token allowance page', () => {
+    const { getByText } = renderWithProvider(
+      <TokenAllowance {...props} />,
+      store,
+    );
+    expect(getByText('Set a spending cap for your')).toBeInTheDocument();
+  });
+
+  it('should render reject button', () => {
+    const { getByTestId } = renderWithProvider(
+      <TokenAllowance {...props} />,
+      store,
+    );
+    const onCloseBtn = getByTestId('page-container-footer-cancel');
+    expect(onCloseBtn).toBeInTheDocument();
+  });
+
+  it('should click View details and show function type', () => {
+    const { getByText } = renderWithProvider(
+      <TokenAllowance {...props} />,
+      store,
+    );
+
+    const viewDetailsButton = getByText('View details');
+    fireEvent.click(viewDetailsButton);
+    expect(getByText('Function: Approve')).toBeInTheDocument();
+  });
+
+  it('should click Use default and set input value to default', () => {
+    const { getByText, getByTestId } = renderWithProvider(
+      <TokenAllowance {...props} />,
+      store,
+    );
+
+    const useDefault = getByText('Use default');
+    fireEvent.click(useDefault);
+
+    const input = getByTestId('custom-spending-cap-input');
+    expect(input.value).toBe('1');
+  });
+
+  it('should call back button when button is clicked and return to previous page', () => {
+    const { getByText, getByTestId } = renderWithProvider(
+      <TokenAllowance {...props} />,
+      store,
+    );
+
+    const textField = getByTestId('custom-spending-cap-input');
+    fireEvent.change(textField, { target: { value: '1' } });
+
+    const nextButton = getByText('Next');
+    fireEvent.click(nextButton);
+
+    const backButton = getByText('< Back');
+    fireEvent.click(backButton);
+
+    expect(getByText('Set a spending cap for your')).toBeInTheDocument();
+  });
+
+  it('should click Verify contract details and show popup Contract details', () => {
+    const { getByText } = renderWithProvider(
+      <TokenAllowance {...props} />,
+      store,
+    );
+
+    const verifyContractDetails = getByText('Verify contract details');
+    fireEvent.click(verifyContractDetails);
+
+    expect(getByText('Contract details')).toBeInTheDocument();
+
+    const gotIt = getByText('Got it');
+    fireEvent.click(gotIt);
+    expect(verifyContractDetails).toBeInTheDocument();
```

Comment:
Above test is not checking for popup. `expect(verifyContractDetails).toBeInTheDocument();` does not looks useful.

Reply:
I updated the test to check if pop-up is closed when `Got it` button is clicked.